### PR TITLE
Rate 215

### PR DESCRIPTION
**Pull request by ux-AylonS**

### CODE REVIEW

File: src/doc-editors/RateLimitRulesEditor.vue [hunk 1]
```diff
@@ -245,12 +245,13 @@
...
           </div>
         </div>
-      </div>
+      </div> connected: {{ havePoliciesConnections }}
       <security-policies-connections
           selectedDocType="ratelimits"
           :selectedDocId="localDoc.id"
           @havePoliciesConnections="emitConnections"
           @go-to-route="emitGoToRoute"
+          @connected="emitConnections"
```
Comment (Aviv-Galmidi) [inline src/doc-editors/RateLimitRulesEditor.vue:254]: You don't emit `connected` in this component

File: src/doc-editors/RateLimitRulesEditor.vue [hunk 2]
```diff
@@ -245,12 +245,13 @@
             </div>
           </div>
         </div>
-      </div>
+      </div> connected: {{ havePoliciesConnections }}
```
Comment (Aviv-Galmidi) [inline src/doc-editors/RateLimitRulesEditor.vue]: What's this text for?

File: src/components/SecurityPoliciesConnections.vue [hunk 3]
```diff
@@ -323,6 +324,7 @@ export default defineComponent({
         this.setEntryDeleteIndex(-1)
         this.getConnectedSecurityPoliciesEntries()
       })
+      this.$emit('have-policies-connections', mapEntry['limit_ids'].length > 1)
```
Comment (Aviv-Galmidi) [inline src/components/SecurityPoliciesConnections.vue:327]: This should be done inside the `.then` promise resolution

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/doc-editors/RateLimitRulesEditor.vue (hunk 1 above)
Comment (ux-AylonS) [reply src/doc-editors/RateLimitRulesEditor.vue:254]: Its amazing how it works with this mistake

File: src/components/SecurityPoliciesConnections.vue (hunk 3 above)
Comment (ux-AylonS) [reply src/components/SecurityPoliciesConnections.vue:327]: when it is inside the page refresh before it had the update
however, this way it works well as it takes care of the re-rendering of the delete button immidiatly,
based on testing.

File: src/doc-editors/RateLimitRulesEditor.vue (hunk 2 above)
Comment (ux-AylonS) [reply src/doc-editors/RateLimitRulesEditor.vue]: was for testing only. deleted

File: src/components/SecurityPoliciesConnections.vue (hunk 3 above)
Comment (Aviv-Galmidi) [reply src/components/SecurityPoliciesConnections.vue:327]: This means that if the update API call fails for some reason, the delete button will still be updated

But we're also deleting the item from the array by using `.splice` before the action is completed

For now that's okay - but it's something we should revisit in the future when we align this behavior around the system to only update UI after a successful API call

File: src/doc-editors/RateLimitRulesEditor.vue [hunk 4]
```diff
@@ -251,6 +251,7 @@
           :selectedDocId="localDoc.id"
           @havePoliciesConnections="emitConnections"
```
Comment (Aviv-Galmidi) [inline src/doc-editors/RateLimitRulesEditor.vue:252]: Can be removed as you added `@have-policies-connections="emitConnections"`
Using kebab-case is preferred

**Review by Aviv-Galmidi — APPROVED** [review]

**Pull request merged**